Added Rig

raystubbs opened the pull request:
Hey @dbohdan, me again.  I won't claim that Rig is especially more complete than it was on my last pull request, but I have been actively working on it and have fixed several issues.  Hope you'll check it out.  Thanks.

dbohdan commented:
Okay, now Rig meets my requirements for the list.  Added in 563409f4f6dd1bf9430c8f89b7fb6522877a1fc9.  Thanks!

The description of shared lookup tables in the README made me curious, but didn't go into enough detail to satisfy that curiosity.  It's worth linking to https://github.com/raystubbs/rig/blob/6070c62eede163c8fb536f393d0f70c3e5ac46a3/kernel/rigK_Record.h#L9 and/or putting a summary of this information in the README.  It would also be interesting to see a concrete comparison of memory usage between the same program implemented in Lua and Rig.

raystubbs commented:
Yeah, the shared indices are a central part of Rig, and the feature should
be highlighted; I just haven't thought of a way to explain them in the
readme without going too far into detail.  But a more comprehensive
explanation is available at https://rig-lang/Language guide/Records.

I did do a simple benchmark to test/demonstrate the advantage of Rig's
records for memory efficiency.  But it isn't a realistic application, just
a microbenchmark for creating a long linked list.  The results of this can
be found at
https://GitHub.com/raystubbs/rig-cli/benchmarks#record-memory-usage.  Most
of the other benchmarks are likely obsolete since I've added a few
performance optimizations since running them; but this one should still be
accurate.

Perhaps adding a link to the language guide's description of records to the
README would be helpful, and wouldn't clutter up the summary with details.




On Sat, Feb 2, 2019, 11:16 AM dbohdan <notifications@github.com wrote:

> Okay, now Rig meets my requirements for the list. Added in 563409f
> <https://github.com/dbohdan/embedded-scripting-languages/commit/563409f4f6dd1bf9430c8f89b7fb6522877a1fc9>.
> Thanks!
>
> The description of shared lookup tables in the README made me curious, but
> didn't go into enough detail to satisfy that curiosity. It's worth linking
> to
> https://github.com/raystubbs/rig/blob/6070c62eede163c8fb536f393d0f70c3e5ac46a3/kernel/rigK_Record.h#L9
> and/or putting a summary of this information in the README. It would also
> be interesting to see a concrete comparison of memory usage between the
> same program implemented in Lua and Rig.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/dbohdan/embedded-scripting-languages/pull/29#issuecomment-459953334>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGwXD58JKGw9U9vgnls9ZYNzZWXD6F0kks5vJWWEgaJpZM4aJUxb>
> .
>

raystubbs commented:
My bad, typed from my phone and go the links wrong.

Description of record mechanics is here:
https://rig-lang.io/LanguageGuide/Records

Record memory usage benchmark is here:
https://github.com/raystubbs/rig-cli/tree/master/benchmarks#record-memory-usage


On Sat, Feb 2, 2019 at 11:58 AM Ray Stubbs <stubbs.ray@gmail.com> wrote:

> Yeah, the shared indices are a central part of Rig, and the feature should
> be highlighted; I just haven't thought of a way to explain them in the
> readme without going too far into detail.  But a more comprehensive
> explanation is available at https://rig-lang/Language guide/Records.
>
> I did do a simple benchmark to test/demonstrate the advantage of Rig's
> records for memory efficiency.  But it isn't a realistic application, just
> a microbenchmark for creating a long linked list.  The results of this can
> be found at
> https://GitHub.com/raystubbs/rig-cli/benchmarks#record-memory-usage.
> Most of the other benchmarks are likely obsolete since I've added a few
> performance optimizations since running them; but this one should still be
> accurate.
>
> Perhaps adding a link to the language guide's description of records to
> the README would be helpful, and wouldn't clutter up the summary with
> details.
>
>
>
>
> On Sat, Feb 2, 2019, 11:16 AM dbohdan <notifications@github.com wrote:
>
>> Okay, now Rig meets my requirements for the list. Added in 563409f
>> <https://github.com/dbohdan/embedded-scripting-languages/commit/563409f4f6dd1bf9430c8f89b7fb6522877a1fc9>.
>> Thanks!
>>
>> The description of shared lookup tables in the README made me curious,
>> but didn't go into enough detail to satisfy that curiosity. It's worth
>> linking to
>> https://github.com/raystubbs/rig/blob/6070c62eede163c8fb536f393d0f70c3e5ac46a3/kernel/rigK_Record.h#L9
>> and/or putting a summary of this information in the README. It would also
>> be interesting to see a concrete comparison of memory usage between the
>> same program implemented in Lua and Rig.
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/dbohdan/embedded-scripting-languages/pull/29#issuecomment-459953334>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AGwXD58JKGw9U9vgnls9ZYNzZWXD6F0kks5vJWWEgaJpZM4aJUxb>
>> .
>>
>

-- 
Regards,
Ray Stubbs
Computer Science
Undergraduate Student
New Mexico State University

dbohdan commented:
>Perhaps adding a link to the language guide's description of records to the README would be helpful, and wouldn't clutter up the summary with details.

This sounds like a good idea to me.  When I saw the "(TODO)Rig’s Record Design" link leading to a 404 page on https://rig-lang.io I thought there wasn't an article about how the records worked.  I went straight to the source code as a result.  I'd link to https://rig-lang.io/LanguageGuide/Records and maybe the benchmark right in the _Features_ section -> _Fast and Efficient_.

raystubbs commented:
Good idea, that's probably the best option to keep things short and
relevent.  Will update that sometime today.  Thanks for the advice.

On Sat, Feb 2, 2019, 12:17 PM dbohdan <notifications@github.com wrote:

> Perhaps adding a link to the language guide's description of records to
> the README would be helpful, and wouldn't clutter up the summary with
> details.
>
> This sounds like a good idea to me. When I saw the "(TODO)Rig’s Record
> Design" link leading to a 404 page on https://rig-lang.io I thought there
> wasn't an article about how the records worked. I went straight to the
> source code as a result. I'd link to
> https://rig-lang.io/LanguageGuide/Records and maybe the benchmark right
> in the *Features* section -> *Fast and Efficient*.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/dbohdan/embedded-scripting-languages/pull/29#issuecomment-459957016>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGwXD-PzKBpDPoFyzdQJZeJtjgIVkX7Cks5vJXOwgaJpZM4aJUxb>
> .
>
